### PR TITLE
fix(web-forms#826): attachment removal on edit submission

### LINKED ISSUE #826: Attached files in media widgets aren't removed in editing
**Problem description**

In media widgets ( audio, video, signature, file, image, draw widgets) the attached file isn't removed after using the X button and confirming the deletion while editing the submission. After reopening the edited submission the deleted files are still present.
The issue occurs both on test.getodk.cloud and staging.getodk.cloud but there is different information in "Submission details" page in "Edited" part.

**Screenshot**

Image

**Steps to reproduce**

1. Create a submission with uploaded media in image, audio, video, file widgets or use the submission nr 1 in
https://staging.getodk.cloud/projects/235/forms/all-widgets/submissions
2. Open the editing submission page.
3. Go to media widgets and click X and confirm deleting the file in the dialog.
4. Send the submission.
5. Check the "submission details" page- on test.getodk.cloud there's no info about the changes; on staging.getodk.cloud the deleted file gets a new name.
6. Open the editing submission page again- check if the media files are removed. 

**Expected behavior**

The deleted media should be removed.

**Web Forms version**
v0.24.0
**Browser** 
Chrome, Firefox

### PR DESCRIPTION
**Pull request by latin-panda**
Closes https://github.com/getodk/web-forms/issues/826

#### Before submitting this PR, please make sure you have:

- [ ] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced
- [ ] run `npm run changeset` to generate a changeset file for changes that should be included in the release notes

### CODE REVIEW

**Comment by latin-panda** [comment]
@garethbowen Could you please review this fix? I did a quick smoke test and everything looks good so far. Tomorrow, I plan to try out some more edge cases, as I ran out of time today. Thanks a lot!

File: packages/xforms-engine/test/integration/instance-attachments/binary-output.test.ts [hunk 1]
```diff
@@ -844,5 +844,46 @@ describe('Instance attachments: binary output', () => {
...
+						model(
+							mainInstance(
+								t('data id="basic-upload-control"',
+									t('file-upload'),
+									t('meta',
+										t('instanceID', FAKE_INSTANCE_ID)))),
+							bind('/data/file-upload').type('binary'))),
+					body(
+						upload('/data/file-upload')))
+      );
```
Comment (garethbowen) [inline packages/xforms-engine/test/integration/instance-attachments/binary-output.test.ts:867]: Whitespace :/

File: packages/xforms-engine/src/lib/reactivity/createInstanceAttachment.ts [hunk 2]
```diff
@@ -101,15 +101,16 @@ const instanceAttachmentState = (
...
     return {
       computedName: null,
       intrinsicName: existingName,
       file: null,
       loading: !!loading,
       loadingError: error ?? false,
-      dirty: false,
+      // writtenAt is set when the client calls setValue(null), meaning the user cleared the attachment.
+      // Mark dirty so the submission payload handles it correctly.
+      dirty: writtenAt != null,
```
Comment (garethbowen) [inline packages/xforms-engine/src/lib/reactivity/createInstanceAttachment.ts]: I think it might be easier to understand what's going on if this code also set the `intrinsicName` to null when `writtenAt != null`. That way the `getFileName` code wouldn't need to check dirty because the filename would already be null.

**Review by garethbowen — APPROVED** [review]
One optional suggestion, and a whitespace nit, but other than that this looks good.

File: packages/xforms-engine/test/integration/instance-attachments/binary-output.test.ts (hunk 1 above)
Comment (latin-panda) [reply packages/xforms-engine/test/integration/instance-attachments/binary-output.test.ts:867]: Fixed!

File: packages/xforms-engine/src/lib/reactivity/createInstanceAttachment.ts (hunk 2 above)
Comment (latin-panda) [reply packages/xforms-engine/src/lib/reactivity/createInstanceAttachment.ts]: Fixed!
